src/player.h

```diff
@@ -293,7 +293,7 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         /** Returns true if the player is wearing an active optical cloak */
         bool has_active_optcloak() const;
         /** Adds a bionic to my_bionics[] */
-        void add_bionic(std::string const &b);
+        void add_bionic( std::string const &b, std::map<body_part, int> affected_bodyparts );
```

Comment:
Unless you need to depend upon a specific ordering you could consider `std::unordered_map` over `std::map`. The former has (possibly irrelevant) better performance characteristics and is semantically clearer.